### PR TITLE
[refactor] Removed Backend.get_<data> methods

### PR DESCRIPTION
**Pull request by lmondada**
I am currently making the necessary changes to pytket-extensions

### CODE REVIEW

**Comment by lmondada** [comment]
We're putting this on hold until we have cleared how exactly we'd like to do this.

**Comment by lmondada** [comment]
I'm now going to update the extensions.

File: pytket/pytket/backends/backend.py
Comment (cqc-alec) [inline pytket/pytket/backends/backend.py:368]:
```suggestion
        Submits a circuit to the backend and returns results
```

File: pytket/pytket/backends/backend.py
Comment (cqc-alec) [inline pytket/pytket/backends/backend.py:374]:
```suggestion
        :py:meth:`Backend.process_circuit` followed by :py:meth:`Backend.get_result`.
```

File: pytket/pytket/backends/backend.py
Comment (cqc-alec) [inline pytket/pytket/backends/backend.py:383]:
```suggestion
        Submits circuits to the backend and returns results
```

File: pytket/pytket/backends/backend.py
Comment (cqc-alec) [inline pytket/pytket/backends/backend.py:389]:
```suggestion
        :py:meth:`Backend.process_circuits` followed by :py:meth:`Backend.get_results`.
```

File: pytket/pytket/backends/backend.py
Comment (cqc-alec) [inline pytket/pytket/backends/backend.py:392]: Why do we need `type: ignore` here?

**Comment by lmondada** [comment]
A draft `pytket-extensions` PR has been opened: https://github.com/CQCL/pytket-extensions/pull/202. When this is merged in, we can run the CI tests on the extensions.

**Review by cqc-alec — CHANGES_REQUESTED** [review]

File: pytket/pytket/backends/backend.py
Comment (lmondada) [reply pytket/pytket/backends/backend.py]: Mypy is a bit fidly with `**kwargs`. I have found another fix without `type: ignore` though.

File: pytket/pytket/backends/backend.py
Comment (lmondada) [reply pytket/pytket/backends/backend.py]: done

File: pytket/pytket/backends/backend.py
Comment (lmondada) [reply pytket/pytket/backends/backend.py]: done

File: pytket/pytket/backends/backend.py
Comment (lmondada) [reply pytket/pytket/backends/backend.py]: done

File: pytket/pytket/backends/backend.py
Comment (lmondada) [reply pytket/pytket/backends/backend.py]: done

**Review by cqc-alec — APPROVED** [review]
Looks good to me; after merging we can make a candidate release to enable the extensions and manual/examples to be updated.

**Pull request merged**
